src/pages/checkout/index.tsx

```diff
@@ -16,47 +18,78 @@ type Address = {
 const Checkout: React.FC = () => {
 	const { isLoggedIn, user } = useAuth();
 	const { loadCartData, cartData } = useCart();
+	const { handleOpenPopup, handleClosePopup } = usePopup();
 	const location = useLocation();
-	const recievedType = location.state?.orderType;
+	const receivedType = location.state?.orderType;
+	const deliveryType = receivedType ? receivedType : 'Point of delivery';
 	const { values, handleChange, errors, isValid } = useFormAndValidation();
 	const navigate = useNavigate();
-
-	const [deliveryType, setDeliveryType] = React.useState<string>('shipment');
 	const [selectedPayment, setSelectedPayment] = React.useState<string>('');
 	const [selectedTime, setSelectedTime] = React.useState<string>('');
 	const [selectedAddress, setSelectedAddress] = React.useState<string>('');
-	const [actualDeliveryType, setActualDeliveryType] = React.useState<string>('');
 	const userAddresses = user?.addresses as unknown[] as Address[];
 	const [comment, setComment] = React.useState<string>('');
+	const [popupText, setPopupText] = useState('');
 	const addressesById = {
-		1: 'Ленина, 23, кв. 19',
-		2: 'Улица 2, дом 5',
-		3: 'Другой адрес и т.д.',
+		1: 'Санкт-Петербург Невский проспект 17',
+		2: 'Санкт-Петербург Горохова улица 10',
+		3: 'Санкт-Петербург Невский проспект 89',
+		4: 'Санкт-Петербург Большой Самсониевский 6',
+		5: 'Санкт-Петербург Лесной проспект 56',
+		6: 'Санкт-Петербург Владимирский проспект 1',
+		7: 'Санкт-Петербург Лиговский проспект 170',
+	};
+
+	const openInfoPopup = (text: string) => {
+		setPopupText(text);
+		handleOpenPopup('openPopupCheckoutResponse');
 	};
 
 	const handleSubmitOrder = () => {
-		if (isLoggedIn) {
-			if (
-				!selectedPayment ||
-				!actualDeliveryType ||
-				(actualDeliveryType !== 'shipment' && !selectedAddress?.toString().trim())
-			) {
-				alert('Пожалуйста, заполните все обязательные поля');
+		switch (true) {
```

Comment:
return и сеттер можно в одну строчку писать)

```suggestion
		switch (true) {
			case isLoggedIn && !user.first_name:
				return openInfoPopup('Пожалуйста, заполните имя в личном кабинете');
			case isLoggedIn && !user.last_name:
				return openInfoPopup('Пожалуйста, заполните фамилию в личном кабинете');

			case isLoggedIn && !user.phone_number:
				return openInfoPopup('Пожалуйста, заполните номер телефона в личном кабинете');

			case isLoggedIn && !selectedAddress?.toString().trim():
				return openInfoPopup(
					'Пожалуйста, выберите адрес (добавить адрес можно в личном кабинете)'
				);
			case isLoggedIn && !selectedPayment:
				return openInfoPopup('Пожалуйста, выберите способ оплаты');
```

Reply:
👍

Reply:
@jsapro, в default можно оставить пустой return)